config/webpack/terra-dev-site.webpack.config.js

```diff
@@ -1,32 +1,92 @@
 
 const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
+const generateAppConfig = require('../../scripts/generate-app-config/generateAppConfig');
+const loadDefaultSiteConfig = require('../../scripts/generate-app-config/loadDefaultSiteConfig');
 const fs = require('fs');
 
-const isFile = filePath => (fs.existsSync(filePath) && !fs.lstatSync(filePath).isDirectory());
+/**
+* Add's an alias and a 'source' alias if not in prod mode and hot reloading is enabled.
+*/
+const addAlias = (acc, name, location, hotReloading, production) => {
+  acc[name] = path.join(location);
+  if (!production && hotReloading.enabled) {
+    acc[`${name}/${hotReloading.dist}`] = `${name}/${hotReloading.source}`;
+  }
+};
+
+/**
+* Aliases all mono-repo packages. This ensures the correct module is hit if the site is hosting an item used to create itself (ouroboros).
+*/
+const aliasMonoRepoPackages = (monoRepoPackageDir, hotReloading, production) => {
+  // If no package directory is found, do nothing.
+  if (!fs.existsSync(monoRepoPackageDir)) {
+    return {};
+  }
+  // For each non hidd directory, create an alias.
+  return fs.readdirSync(monoRepoPackageDir).reduce((acc, packageName) => {
+    // ignore any hidden files
+    if (packageName[0] !== '.') {
+      addAlias(acc, packageName, path.join(monoRepoPackageDir, packageName), hotReloading, production);
+    }
+    return acc;
+  }, {});
+};
+
+/**
+* Alias the current package, Mostlikely this isn't neaded, but doesn't hurt if someone is referencing files oddly.
```

Comment:
Mostlikely --> most likely

Think we should still include?

Reply:
Good point, my only use for this now is toolkit itself.